src/BloomExe/Edit/EditingView.cs

```diff
@@ -727,15 +729,20 @@ private void OnChangeImage(DomEventArgs ge)
 			}
 			using (var dlg = new ImageToolboxDialog(imageInfo, null))
 			{
-				if (DialogResult.OK == dlg.ShowDialog())
+				var result = dlg.ShowDialog();
+				// Check memory for the benefit of developers.  The user won't see anything.
+				Palaso.UI.WindowsForms.Reporting.MemoryManagement.CheckMemory(true, "picture chosen or canceled", false);
+				if (DialogResult.OK == result)
 				{
-
 					// var path = MakePngOrJpgTempFileForImage(dlg.ImageInfo.Image);
 					SaveChangedImage(imageElement, dlg.ImageInfo, "Bloom had a problem including that image");
+					// Warn the user if we're starting to use too much memory.
+					Palaso.UI.WindowsForms.Reporting.MemoryManagement.CheckMemory(true, "picture chosen and saved", true);
 				}
 			}
 			Logger.WriteMinorEvent("Emerged from ImageToolboxDialog Editor Dialog");
 			Cursor = Cursors.Default;
+			imageInfo.Dispose();	// ensure memory doesn't leak
```

Comment:
The imageInfo is created in two different ways, which makes a using block infeasible.  I did consider a try finally setup, but never got around to implementing it.  (possibly never thought it important enough?)